### PR TITLE
CB-8198 Unified console output logic for core platforms

### PR DESCRIPTION
**Pull request by daserge**
[Proposal: Unified logging on cordova-discuss](https://github.com/cordova/cordova-discuss/issues/14)
Added advanced ansi-based logger
Redirected events to logger

[Corresponding Lib updates PR](https://github.com/apache/cordova-lib/pull/306)
[Jira issue](https://issues.apache.org/jira/browse/CB-8198)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,10 +29,11 @@
     "cli"
   ],
   "dependencies": {
-    "cordova-lib": "5.3.1",
-    "q": "1.0.1",
+    "ansi": "^0.3.0",
+    "cordova-lib": "5.2.0",
```
Comment (nikhilkh) [inline package.json]: It does not see correct to change this version.

File: src/logger.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    return message;
+}
+
+logger.log = function (logLevel, message) {
+    if (this.levels[logLevel] >= this.levels[this.logLevel]) {
+        var isVerbose = this.logLevel === 'verbose';
+        var prefix = this.prefixes[logLevel] ? this.prefixes[logLevel] + ': ' : '';
+        suffix = '\n';
```
Comment (nikhilkh) [inline src/logger.js]: This should be require('OS').newline

File: src/logger.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+        if(message instanceof Error) {
+            message = formatError(message, isVerbose);
+        }
+
+        message = prefix + message + suffix;
```
Comment (nikhilkh) [inline src/logger.js]: Lot of current error messages (at least in platforms) start with the 'Error:' prefix. I propose we strip that out from the beginning of the message to ensure we do not have double prefixes. In the long run we should change the platform code not to have the prefix.

File: src/logger.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        }
+        var color = this.colors[logLevel];
+        !!color && this.cursor.bold().fg[color]();
+        this.cursor.write(message);
+        this.cursor.reset();
+    }
+};
+
+logger.addLevel = function (level, severity, prefix, color) {
+    this.levels[level] = severity;
```
Comment (nikhilkh) [inline src/logger.js]: It's odd that adding the level, ends up setting the current level to that value.

Comment (daserge) [inline src/logger.js]: I believe it creates a dictionary of level-severity correspondence actually.
Log level is set up in `setLevel`.

Comment (nikhilkh) [inline src/logger.js]: sorry - my bad.

File: src/logger.js (hunk 3 above)
Comment (daserge) [inline src/logger.js]: Makes sense.

File: package.json (hunk 1 above)
Comment (daserge) [inline package.json]: Sorry, it's an overlook from my side.

File: src/logger.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    var message = '';
+    if(isVerbose) {
+        message = error.stack;
+    } else {
+        // Some error messages start with 'Error: ' prefix, so cut it off here
+        // TODO: Update platforms code to remove such prefixes
+        message = error.message && error.message.replace(/^error:\s+/i, '');
+    }
+
+    if(error instanceof CordovaError && error.code !== CordovaError.UNKNOWN_ERROR) {
```
Comment (nikhilkh) [inline src/logger.js]: Consider refactoring this in CordovaError

**Pull request closed without merging**